internal/resource/config/virtual_host_names_data_source.go

```diff
@@ -0,0 +1,111 @@
+package config
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	client "github.com/pingidentity/pingfederate-go-client/v1125/configurationapi"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/datasource/common/id"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &virtualHostNamesDataSource{}
+	_ datasource.DataSourceWithConfigure = &virtualHostNamesDataSource{}
+)
+
+// VirtualHostNamesDataSource is a helper function to simplify the provider implementation.
+func NewVirtualHostNamesDataSource() datasource.DataSource {
+	return &virtualHostNamesDataSource{}
+}
+
+// virtualHostNamesDataSource is the datasource implementation.
+type virtualHostNamesDataSource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type virtualHostNamesDataSourceModel struct {
+	Id               types.String `tfsdk:"id"`
+	VirtualHostNames types.List   `tfsdk:"virtual_host_names"`
+}
+
+// GetSchema defines the schema for the datasource.
+func (r *virtualHostNamesDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	schemaDef := schema.Schema{
+		Description: "Manages VirtualHostNames.",
```

Comment:
"Manages" probably isn't the language we should use on data sources. I used "Describes" for data sources in PD

Reply:
Fixed